File: binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    HashMap<Integer, Integer> map = new HashMap<>();
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> view = new ArrayList<>();
        if (root == null) return view;
        dfs(root, 0);
        int maxLevel = Collections.max(map.keySet());
        for (int i = 0; i <= maxLevel; i++) {
            view.add(map.get(i));
        }
        return view;
    }

    private void dfs(TreeNode root, int level) {
        if (root == null) {
            return;
        }
        
        map.put(level, root.val);
        level++;
        dfs(root.left, level);
        dfs(root.right, level);
    }
}

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        view = {}
        def dfs(root, height):
            if not root:
                return
            view[height] = root.val
            dfs(root.left, height + 1)
            dfs(root.right, height + 1)
        dfs(root, 0)
        return list(view.values())
